### DISCUSSION TITLE
Local Supabase migration setup issue

### DISCUSSION THREAD

**Discussion by zakrzaq** [supabase/supabase#12766] [Questions]
I have:
logged in `supbase login`
setup project `supabase link --project-ref xxxxx`

yet no `remote_commit` can be found in `supabase/migrations/`

when running `supabase db remote commit` I get:

```
Error: The remote database's migration history is not in sync with the contents of supabase/migrations. Resolve this by:
- Updating the project from version control to get the latest supabase/migrations,
- Pushing unapplied migrations with supabase db push,
- Or failing that, manually editing supabase_migrations.schema_migrations table with supabase migration repair.
Try rerunning the command with --debug to troubleshoot the error.
```
I tired `supabase db push` before running `supbase remote commit` as suggested in some comments but it results in same error

**Comment by caseybaggz** [comment]
I was able to fix this by going into the project > Tables > selecting the `supabase_migrations` option from the select then deleting anything that is listed in there. 🤷‍♀️

**Reply by silentworks** [reply]
In the latest CLI it gives even better error reporting and instructions on how you can fix this from the CLI itself.

**Comment by heblol** [comment]
I have the same, applying `supabase db reset` and then `supabase db pull ` results in the same error. Also, the CLI is not very informative. I even deleted the local repository and still have the same error? `supabase db diff` says `"No schema changes found"`

**Comment by ahmadbilaldev** [comment]
I have the same issue. I have migrations on the remote that I want to pull in my local db, but `supabase db pull` returns the error above. What is the point of `pull` when it won't pull remote changes?

**Reply by Nelhoa** [reply]
Did you find a solution ? Stuck with the same issue…

**Comment by Rubenburdin** [comment]
The supabase db remote commit command was deprecated and removed in newer CLI versions. If youre on a recent CLI the equivalent workflow is now supabase db pull which generates a migration file from your remote schema. But the underlying problem youre hitting is the same, the migration history table on remote has entries that dont match whats in your local supabase/migrations folder.

The cleanest fix when starting fresh is what caseybaggz mentioned, clear the supabase_migrations.schema_migrations table on remote, then run supabase db pull to create a baseline migration from your current remote schema. This gives you a clean starting point. Just be aware this means youre saying "forget all previous migration history and start from the current state."

For ahmadbilaldev and Nelhoa's question about why pull wont pull remote changes, the command isnt designed to sync migration files themselves. It generates a new migration based on the difference between your local migrations and the remote schema. If theyre already in sync schema wise but the history table is mismatched, you need to use supabase migration repair to mark migrations as applied or reverted without actually running them (https://supabase.com/docs/reference/cli/supabase-migration-repair).

Running supabase db pull --debug should show you exactly which migrations are causing the mismatch. From there you can either repair the history or do a clean reset depending on your situation.

For teams where the bigger headache is keeping actual data consistent between their Supabase database and external systems like HubSpot or Salesforce rather than schema migrations, thats a different problem entirely and more what Stacksync addresses. But for CLI migration sync issues the repair command or history reset should get you moving again.
